lib/cache/entry.js

```diff
@@ -143,6 +138,12 @@ class CacheEntry {
         return entryA.policy.satisfies(entryB.request)
       }, {
         validateEntry: (entry) => {
+          // clean out entries with a buggy content-encoding value
+          if (entry.metadata &&
+              entry.metadata.resHeaders &&
+              entry.metadata.resHeaders['content-encoding'] === null)
+            return false
+
```

Comment:
kind of a nitpick, but what do you think about making this use `||` instead of `&&` between each check? that way if, somehow, someone ends up with an entry that doesn't have `metadata` or `resHeaders` that would also get cleared out as invalid

Reply:
Hmm, this is what you mean?

```js
          if (!entry.metadata ||
              !entry.metadata.resHeaders ||
              entry.metadata.resHeaders['content-encoding'] === null)
```

That sounds reasonable, but I think we'll have to update a few tests to more thoroughly mock cache entries. If I remember correctly, I had initially just had the `=== null` check, but that ran afoul of tests which omitted `metadata.resHeaders`, so I added the other checks.

Reply:
I guess if we start going down that route, it could make sense to more systematically validate everything the code assumes to be in the entry, like `reqHeaders` and `url` and stuff too? If that's the case, then I think that'd be better in a separate PR.

Reply:
you're absolutely right, i think more actual validation is what we really want, and it does make way more sense in a separate PR